### PR TITLE
Localization support added for messages

### PR DESCRIPTION
**Pull request by LoekJanssen**
I created localization in German, Dutch and English for the messages in
the top messagebar and some logs.

I've used full English strings for lookup.

This is the new version of PR #548, because I made a few mistakes.

### CODE REVIEW

**Comment by samhed** [comment]
The commits are very messy, could you clean them up a bit?

**Comment by LoekJanssen** [comment]
@samhed I've done a rebase and put everything in one commit

**Comment by samhed** [comment]
Thank you!

File: app/locale/locale-de-DE.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Language = {
+	"Connected (encrypted) to ": "Verbunden mit (verschlüsselt) ",
+	"Connected (unencrypted) to ": "Verbunden mit (unverschlüsselt) ",
+	"Must set host and port": "Richten Sie Host und Port ein",
+	"Disconnect timeout":	"Timeout beim trennen",
```
Comment (samhed) [inline app/locale/locale-de-DE.js]: Nitpick: Remove the extra spaces after the ":" here please :)

Comment (samhed) [inline app/locale/locale-de-DE.js]: Same thing for en-GB and nl-NL

File: app/ui.js [hunk 2]
```diff
@@ -475,8 +475,8 @@ var UI;
                     statusElem.classList.add("noVNC_status_normal");
                     break;
             }
-
-            statusElem.innerHTML = text;
+
```
Comment (samhed) [inline app/ui.js]: Remove this whitespace change please

**Comment by samhed** [comment]
I didn't mean you should remove the spaces after ALL the colons :smiley: 

There were extra spaces after the colon on the "Disconnect timeout" lines!

**Comment by LoekJanssen** [comment]
Ah haha, I saw that, but I thought you meant all. I will change it now :)

File: app/ui.js [hunk 3]
```diff
@@ -476,7 +476,7 @@ var UI;
                     break;
             }
 
-            statusElem.innerHTML = text;
+            statusElem.innerHTML = Util.Localisation.get(text);
```
Comment (CendioOssman) [inline app/ui.js]: This is generally not a good idea as it makes it very difficult to find what needs to be translated. Only put translations on literals.

File: app/ui.js [hunk 4]
```diff
@@ -369,9 +369,9 @@ var UI;
                 case 'connected':
                     UI.connected = true;
                     if (rfb && rfb.get_encrypt()) {
-                        msg = "Connected (encrypted) to " + UI.desktopName;
+                        msg = Util.Localisation.get("Connected (encrypted) to ") + UI.desktopName;
```
Comment (CendioOssman) [inline app/ui.js]: A good start perhaps, but this won't work long term. What if the order of words needs to be changed? A printf() style API is need to handle languages outside the Germanic family.

File: core/util.js [hunk 5]
```diff
@@ -366,4 +366,85 @@ Util.Flash = (function () {
...
+        'nl':    'nl-NL',
+        'nl-NL': 'nl-NL',
+        'nl-BE': 'nl-NL',
+        'de':    'de-DE',
+        'de-DE': 'de-DE'
+    },
+
+    // Get language file location
+    getLanguageFileLocation: function () {
+        return '../app/locale/locale-'+Util.Localisation.getLanguageCode()+'.js';
```
Comment (CendioOssman) [inline core/util.js]: Util should only be used for core API things and should not reference the app side of things.

Comment (samhed) [inline core/util.js]: You mean that we can't expect everyone to have an 'app' or a 'app/locale' directory I guess.. Good point, didn't think of that!

File: app/ui.js (hunk 3 above)
Comment (samhed) [inline app/ui.js]: Fixed in fdb02f857083ab1015ccaaa8d0d1be2a978537a3

**Pull request merged**